### PR TITLE
fix(brew): relocate text in skip-relocation bottles

### PR DESCRIPTION
**Pull request by jdx**
## Summary

- always run text placeholder relocation for Homebrew bottles
- treat `:any_skip_relocation` as skipping only binary linkage relocation
- add regression coverage proving text files are changed while Mach-O content remains untouched

## Root cause

The brew pour path skipped `relocate_keg` entirely for macOS bottles tagged `:any_skip_relocation`. Homebrew uses that tag only to suppress dynamic linkage relocation; it still replaces text placeholders such as `@@HOMEBREW_PREFIX@@` and `@@HOMEBREW_CELLAR@@`.

This left scripts and configuration files containing unresolved placeholders and produced an empty `changed_files` receipt entry.

Reported in https://github.com/jdx/mise/discussions/11656.

## Validation

- `cargo test system::packages::brew::relocate::tests` (8 passed)
- `mise run format`
- `git diff --check`

*AI-assisted — Tool: Codex; model: unavailable/unavailable; version: unavailable.*

---

> [!NOTE]
> **Medium Risk**
> Touches the Homebrew bottle pour/relocation path used on every install; behavior change is intentional and covered by a regression test, but wrong linkage skipping could still break binaries on edge-case bottles.
> 
> **Overview**
> **Fixes `:any_skip_relocation` bottles leaving `@@HOMEBREW_*@@` placeholders in scripts and config** by always running relocation during pour instead of skipping it entirely on macOS (and on Linux when the bottle was built by Homebrew ≥ 5.1.15).
> 
> `relocate_keg` now takes a **`skip_linkage`** flag: when set, Mach-O, ELF, and other binary linkage paths are left unchanged, but **text files and shebang-only executables** still get placeholder replacement—matching Homebrew’s meaning of `:any_skip_relocation`. The pour path always invokes relocation and passes `skip_linkage` from the bottle cellar tag (with the existing Linux Homebrew version gate for mis-tagged older bottles).
> 
> Adds **`test_skip_linkage_still_relocates_text_files`** to assert text is rewritten, Mach-O bytes and permissions stay untouched, and receipts list only changed text files.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 08427d30b574a39b1e36a2ba7d8bd07a89db513e. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved bottle relocation to preserve text placeholder replacements while safely skipping binary linkage updates when required.
  - Prevented unintended changes to Mach-O, ELF, and other binary files during relocation.
  - Continued relocating text files and shebang-based executables as expected.
  - Updated relocation results to report only files that were actually modified.
  - Preserved file permissions during relocation and improved compatibility across supported Homebrew and Linux environments.

### CODE REVIEW

File: src/system/packages/brew/relocate.rs [hunk 1]
```diff
@@ -221,6 +235,9 @@ pub fn relocate_keg(keg: &Path, formula_name: &str) -> Result<RelocationReport>
         let macho = is_macho(&content);
         let elf = cfg!(target_os = "linux") && super::elf::is_elf(&content);
         let shebang_end = text_executable_shebang_end(&content);
+        if skip_linkage && (macho || elf || (content.contains(&0) && shebang_end.is_none())) {
+            continue;
```
[the remark this answers is not in the document]
Comment (jdx) [reply src/system/packages/brew/relocate.rs]: Fixed in e1ee3c9ee. The skip decision now happens before permissions are relaxed, and the regression test verifies a read-only skipped binary retains mode `0444`.

*AI-assisted — Tool: Codex; model: unavailable/unavailable; version: unavailable.*
